loader/loader_windows.c

```diff
@@ -802,121 +801,119 @@ VkResult windows_read_sorted_physical_devices(struct loader_instance *inst, stru
+            if (vkres != VK_SUCCESS) {
+                loader_instance_heap_free(inst, sorted_array[*sorted_devices_count].physical_devices);
+                sorted_array[*sorted_devices_count].physical_devices = NULL;
+                if (vkres == VK_ERROR_OUT_OF_HOST_MEMORY) {
                     res = VK_ERROR_OUT_OF_HOST_MEMORY;
                     goto out;
-                } else if (vkres != VK_SUCCESS) {
-                    loader_log(inst, VULKAN_LOADER_WARN_BIT, 0,
-                               "Failed to convert DXGI adapter into Vulkan physical device with unexpected error code");
+                } else {
+                    loader_log(inst, VULKAN_LOADER_WARN_BIT, 0, "Failed to convert DXGI adapter into Vulkan physical device");
                     continue;
                 }
-
-                // Get the actual physical devices
-                if (0 != count) {
-                    do {
-                        sorted_array[*sorted_count].physical_devices =
-                            loader_instance_heap_realloc(inst, sorted_array[*sorted_count].physical_devices,
-                                                         sorted_array[*sorted_count].device_count * sizeof(VkPhysicalDevice),
-                                                         count * sizeof(VkPhysicalDevice), VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
-                        if (sorted_array[*sorted_count].physical_devices == NULL) {
-                            res = VK_ERROR_OUT_OF_HOST_MEMORY;
-                            break;
-                        }
-                        sorted_array[*sorted_count].device_count = count;
-                    } while ((vkres = icd_term->scanned_icd->EnumerateAdapterPhysicalDevices(
-                                  icd_term->instance, description.AdapterLuid, &count,
-                                  sorted_array[*sorted_count].physical_devices)) == VK_INCOMPLETE);
-                }
-
-                if (vkres != VK_SUCCESS) {
-                    loader_instance_heap_free(inst, sorted_array[*sorted_count].physical_devices);
-                    sorted_array[*sorted_count].physical_devices = NULL;
-                    if (vkres == VK_ERROR_OUT_OF_HOST_MEMORY) {
-                        res = VK_ERROR_OUT_OF_HOST_MEMORY;
-                        goto out;
-                    } else {
-                        loader_log(inst, VULKAN_LOADER_WARN_BIT, 0, "Failed to convert DXGI adapter into Vulkan physical device");
-                        continue;
-                    }
-                }
-                inst->total_gpu_count += (sorted_array[*sorted_count].device_count = count);
-                sorted_array[*sorted_count].icd_index = icd_idx;
-                sorted_array[*sorted_count].icd_term = icd_term;
-                ++(*sorted_count);
             }
-
-            adapter->lpVtbl->Release(adapter);
+            sorted_array[*sorted_devices_count].device_count = count;
```

Comment:
Aren't you losing the actual device count here?  Since one selection may have multiple devices?

Reply:
No, because the calling function adds up the devices on its own.
```c
    for (uint32_t i = 0; i < windows_sorted_devices_count; ++i) {
        new_phys_devs_count += windows_sorted_devices_array[i].device_count;
    }
```